qt/main.cpp

```diff
@@ -214,17 +213,8 @@ int main(int argc, char * argv[])
     fmt.setSwapBehavior(QSurfaceFormat::DoubleBuffer);
     fmt.setSwapInterval(1);
     fmt.setDepthBufferSize(16);
-#if defined(OMIM_OS_LINUX)
-    // This is a workaround for older distros, which rely on X.org and Mesa,
-    // where somehow the Mesa driver itself doesn't
-    // make all the otherwise supported GLSL versions available by default
-    // and such requests are somehow disregarded at later stages of execution.
-    // This setting here will be potentially overwritten and overruled anyway,
-    // and only needed to ensure that we have the needed GLSL compiler available
-    // later when we need it.
-    if (app.platformName() == QString("xcb"))
-      fmt.setVersion(3, 2);
-#endif
+    fmt.setProfile(QSurfaceFormat::CoreProfile);
+    fmt.setVersion(3, 2);
```

Comment:
```suggestion
    if (app.platformName() == QString("wayland"))
    {
      // TODO: Check if there is a way to use OpenGL 3.2 on wayland platform.
      fmt.setProfile(QSurfaceFormat::CompatibilityProfile);
      fmt.setVersion(2, 1);
    }
    else
    {
      // TODO: Avoid copy-pasting the same version/profile in map_widget.cpp and test_main_loop.cpp
      fmt.setProfile(QSurfaceFormat::CoreProfile);
      fmt.setVersion(3, 2);
    }
```

@Ferenc- can you please test if this PR with the patch above will work on wayland?

Reply:
@Ferenc- @vng did you test this case? I don't like that there is a _default_ _undefined_ behavior on some platforms. Can you please try this approach with this patch instead of the merged one?